core/src/integration/java/com/radixdlt/integration/steady_state/deterministic/consensus/DivergentExecutionLivenessBreakTest.java

```diff
@@ -0,0 +1,293 @@
+ * permissions under this License.
+ */
+
+package com.radixdlt.integration.steady_state.deterministic.consensus;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import com.radixdlt.consensus.LedgerHashes;
+import com.radixdlt.consensus.bft.Round;
+import com.radixdlt.consensus.vertexstore.ExecutedVertex;
+import com.radixdlt.consensus.vertexstore.VertexStoreState;
+import com.radixdlt.crypto.HashUtils;
+import com.radixdlt.crypto.Hasher;
+import com.radixdlt.environment.EventDispatcher;
+import com.radixdlt.environment.deterministic.network.MessageSelector;
+import com.radixdlt.harness.deterministic.DeterministicTest;
+import com.radixdlt.harness.deterministic.PhysicalNodeConfig;
+import com.radixdlt.harness.predicates.NodePredicate;
+import com.radixdlt.harness.predicates.NodesPredicate;
+import com.radixdlt.ledger.LedgerExtension;
+import com.radixdlt.ledger.LedgerUpdate;
+import com.radixdlt.ledger.RoundDetails;
+import com.radixdlt.ledger.StateComputerLedger;
+import com.radixdlt.mempool.MempoolAdd;
+import com.radixdlt.modules.FunctionalRadixNodeModule;
+import com.radixdlt.modules.FunctionalRadixNodeModule.ConsensusConfig;
+import com.radixdlt.modules.FunctionalRadixNodeModule.NodeStorageConfig;
+import com.radixdlt.modules.StateComputerConfig;
+import com.radixdlt.monitoring.Metrics;
+import com.radixdlt.p2p.NodeId;
+import com.radixdlt.statecomputer.StatelessComputer;
+import com.radixdlt.statecomputer.StatelessTransactionVerifier;
+import com.radixdlt.transactions.RawNotarizedTransaction;
+import com.radixdlt.utils.KeyComparator;
+import java.util.List;
+import java.util.Random;
+import java.util.function.Consumer;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+/**
+ * Tests that a liveness break caused by divergent vertex execution (timeout quorums can still be
+ * formed) doesn't cause the vertex store to grow uncontrollably.
+ */
+public final class DivergentExecutionLivenessBreakTest {
+  private static final int NUM_NODES = 4;
+  private static final int VERTEX_STORE_MAX_SIZE = 50;
+
+  private static final long FIRST_DIVERGENT_ROUND = 20;
+  private static final long LAST_DIVERGENT_ROUND = 200;
+
+  private static final Random random = new Random(123456);
+
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  private DeterministicTest createTest() {
+    return DeterministicTest.builder()
+        .addPhysicalNodes(
+            PhysicalNodeConfig.createSortedBatchWithFakeAddresses(
+                NUM_NODES, KeyComparator.instance()))
+        .messageSelector(MessageSelector.randomSelector(random))
+        .overrideWithIncorrectModule(
+            new AbstractModule() {
+              @Provides
+              @Singleton
+              private StateComputerLedger.StateComputer stateComputer(
```

Comment:
Does this somehow replace the StateComputer below? Why do we need this? To turn off verification somehow?

Could we add a comment explaining how this is supposed to work?